### PR TITLE
[prover] Update Boogie to 2.15.8

### PR DESCRIPTION
**Pull request by wrwg**
### Description

Hopefully this fixes the issues with vanishing model and trace. If so we can enforce in the move repo to use this version.

### Test Plan

NA

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4671)

### CODE REVIEW

File: scripts/dev_setup.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ VAULT_VERSION=1.5.0
 Z3_VERSION=4.11.0
```
Comment (junkil-park) [inline scripts/dev_setup.sh:23]: Z3 also recently has a newer version: https://github.com/Z3Prover/z3/releases/tag/z3-4.11.2

The Z3 version seems unrelated to the Prover issue because upgrading Boogie version itself fixes the Prover issue. However, it may be good to upgrade the Z3 version as well.

Comment (wrwg) [reply scripts/dev_setup.sh:23]: I do not went to set a new z3 version before we know this actually fixes it. We have no idea what the issue is, and whether it is in boogie or Z3.

**Review by movekevin — APPROVED** [review]

**Review by meng-xu-cs — APPROVED** [review]
